src/main/scala/is/hail/utils/Py4jUtils.scala

```diff
@@ -23,7 +26,18 @@ trait Py4jUtils {
       list.add(elem)
     list
   }
-  
+
+  // FIXME: Currently this can only send much less than the limit of
+  // BlockMatrix.toLocalMatrix. That's probably fine.
+  def doubleBDMToBytes(mat: BDM[Double]): Array[Byte] = {
+    // assumes matrix is written as written by BlockMatrix
```

Comment:
BlockMatrix assumes offset = 0 and stride is minimal. But you do need to pass through the transpose state through as well, as done for read and write in RichDenseMatrixDouble, and then use that bit to decide whether to transpose on the numpy side.

Reply:
add an fatal checking that `8 + 8 * mat.rows * mat.cols` doesn't overflow

Reply:
Also, please rename function `bdmToBytes`, as it's clear that it's doubles from the signature. And rename `mat` to the more specific `bdm` that's used elsewhere.
I'm also wary about not checking these assumptions. I suggest moving
`assertCompatibleLocalMatrix` out of BlockMatrix to the RichDenseMatrixDouble class, and changing the name to `assertCompatible()`?

Reply:
I'm putting these in as asserts and moving the actual error handling to the python side.

I think we could also make this handle the offset != 0 and stride != col/row case pretty easily, but that might not be necessary for our use case. What do you think?

Reply:
I think we shouldn't add that complexity unless/until we need it.